merkle-tree/bounded-vec/src/lib.rs

```diff
@@ -819,6 +838,201 @@ mod test {
         }
     }
 
+    fn bounded_vec_from_array<const N: usize>() {
+        let mut rng = thread_rng();
+
+        let arr: [u64; N] = array::from_fn(|_| rng.gen());
+        let vec = BoundedVec::from_array(&arr);
+
+        assert_eq!(&arr, vec.as_slice());
+    }
+
+    #[test]
+    fn test_bounded_vec_from_array_256() {
+        bounded_vec_from_array::<256>()
+    }
+
+    #[test]
+    fn test_bounded_vec_from_array_512() {
+        bounded_vec_from_array::<512>()
+    }
+
+    #[test]
+    fn test_bounded_vec_from_array_1024() {
+        bounded_vec_from_array::<1024>()
+    }
+
+    #[test]
+    fn test_bounded_vec_from_slice() {
+        let mut rng = thread_rng();
+
+        for capacity in 0..10_000 {
+            let vec: Vec<u64> = (0..capacity).map(|_| rng.gen()).collect();
+            let bounded_vec = BoundedVec::from_slice(&vec);
+
+            assert_eq!(vec.as_slice(), bounded_vec.as_slice());
+        }
+    }
+
+    #[test]
+    fn test_bounded_vec_is_empty() {
+        let mut rng = thread_rng();
+        let mut vec = BoundedVec::with_capacity(1000);
+
+        assert!(vec.is_empty());
+
+        for _ in 0..1000 {
+            let element: u64 = rng.gen();
+            vec.push(element).unwrap();
+
+            assert!(!vec.is_empty());
+        }
+    }
+
+    #[test]
+    fn test_bounded_vec_get() {
+        let mut vec = BoundedVec::with_capacity(1000);
+
+        for i in 0..1000 {
+            vec.push(i).unwrap();
+        }
+
+        for i in 0..1000 {
+            assert_eq!(vec.get(i), Some(&i));
+        }
+        for i in 1000..10_000 {
+            assert!(vec.get(i).is_none());
+        }
+    }
+
+    #[test]
+    fn test_bounded_vec_get_mut() {
+        let mut vec = BoundedVec::with_capacity(1000);
+
+        for i in 0..1000 {
+            vec.push(i).unwrap();
+        }
+
+        for i in 0..1000 {
+            let element = vec.get_mut(i).unwrap();
+            assert_eq!(element, &i);
+            *element = i * 2;
+        }
+        for i in 0..1000 {
+            assert_eq!(vec.get_mut(i), Some(&mut (i * 2)));
+        }
+        for i in 1000..10_000 {
+            assert!(vec.get_mut(i).is_none());
```

Comment:
Just saw that capacity is just 1000, this tests only values outside of capacity to return None. Should we also test that values within capacity can be None?

Reply:
Good idea, will do